allennlp/modules/augmented_lstm.py

```diff
@@ -0,0 +1,171 @@
+from typing import Optional, Tuple
+import torch
+from torch.autograd import Variable
+from torch.nn.utils.rnn import pad_packed_sequence, pack_padded_sequence, PackedSequence
+
+from allennlp.common.tensor import get_dropout_mask
+from allennlp.common.checks import ConfigurationError
+
+
+class AugmentedLstm(torch.nn.Module):
```

Comment:
Should this be inside `seq2seq_encoders`, and inherit from `Seq2SeqEncoder`?  Or should we wrap this as either a seq2seq or seq2vec encoder, like we do for the pytorch-native ones?